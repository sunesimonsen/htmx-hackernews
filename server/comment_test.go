package server

import (
	"testing"

	"github.com/sunesimonsen/htmx-hackernews/model"
)

func TestComment(t *testing.T) {
	comment := model.Comment{
		By:     "CapitalistCartr",
		Id:     37176230,
		Parent: 37175721,
		Text:   "The link seems to be Slashdotted into oblivion.",
		Time:   1692369360,
		Kids:   []int{42, 24},
	}

	t.Run("snapshot GET /comment/37176230 with a succesful response", func(t *testing.T) {
		snapshotRequest(t, "/comment/37176230", "/v0/item/37176230.json", comment)
	})

	t.Run("snapshot GET /parts/comment/37176230 with a succesful response", func(t *testing.T) {
		snapshotRequest(t, "/parts/comment/37176230", "/v0/item/37176230.json", comment)
	})

	t.Run("GET /comment/37176230 with a succesful response", func(t *testing.T) {
		response := renderResponse(t, "/comment/37176230", "/v0/item/37176230.json", comment)
		doc := newDocument(response)
		assertSelectedText(t, doc, ".comment-body", comment.Text)

		assertNotMatching(t, doc, "a[href=\"/comment/37176230\"]")
		assertMatching(t, doc, ".comments")
		assertMatching(t, doc, "article[hx-get=\"/parts/comment/42\"]")
	})

	t.Run("GET /parts/comment/37176230 with a succesful response", func(t *testing.T) {
		response := renderResponse(t, "/parts/comment/37176230", "/v0/item/37176230.json", comment)
		doc := newDocument(response)
		assertSelectedText(t, doc, ".comment-body", comment.Text)

		assertMatching(t, doc, "a[href=\"/comment/37176230\"]")
		assertNotMatching(t, doc, ".comments")
	})
}
