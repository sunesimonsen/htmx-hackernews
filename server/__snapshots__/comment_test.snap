
[TestGetComment/with_a_succesful_response - 1]
 
<html lang="en">
  <head>
    <title>Hackernews</title>
    <meta charset="UTF-8"/>
    <meta name="description" content="Hackernews example app in htmx and Go" lang="en"/>
    <link rel="icon" href="/assets/favicon.ico" type="image/x-icon">
    <script src="https://unpkg.com/htmx.org@1.9.4" integrity="sha384-zUfuhFKKZCbHTY6aRR46gxiqszMk5tcHjsVFxnUo8VMus4kHGVdIYVbOYYNlKmHV" crossorigin="anonymous"></script>
    <script src="https://unpkg.com/ulid@2.3.0/dist/index.umd.js"></script>
    <script src="/assets/scripts.js"></script>
    <link rel="stylesheet" href="/assets/main.css">
  </head>
  <body>
    <main>
      
<header class="topBar">
  
<a class="logo" href="/">
    <img src="/assets/y18.gif" />
    <span class="logo-caption">Hacker News</span>
</a>

</header>

      <section class="content">
        
  <article class="comment expanded">
      The link seems to be Slashdotted into oblivion.
      <footer>
          <span>CapitalistCartr</span>
          <span data-timestamp>1692369360</span>
          
      </footer>
      
      <section class="comments">
        
      </section>
      
  </article>

      </section>
    </main>
  </body>
</html>
 
---
